File: 392-is-subsequence/is-subsequence.ts
function isSubsequence(s: string, t: string): boolean {
   if(s.length > t.length) return false;
   let sP = 0;
   for(let i = 0; i < t.length; i ++){
    if(t.charAt(i) === s[sP]) sP ++;
   }
   return (sP === s.length)
};

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if(s.length() == 0) return true;
        if(s.length() > t.length()) return false;
        int sp = 0;
        for(int i = 0; i < t.length(); i ++) {
            if(sp<s.length() && t.charAt(i) == s.charAt(sp)) sp ++;
        }
        return sp == s.length();
    }
}